R/util.R

```diff
@@ -706,6 +778,65 @@ util.find_crypt_paths <- function (files_to_load, initial_path = NA,
 }
 
 
+util.custom_fread <- function(path, ...){
+  # setting colClasses to character prevents non-lossless coersion of column types by fread;
+  # however, non-character types must be converted back to original types afterwards
+  fread(
+    input = path,
+    data.table = FALSE,
+    sep = ",",
+    na.strings = c("NA", ""),
+    colClasses = "character",
+    ...
+  ) %>%
+    util.to_ascii %>%
+    util.apply_columns(., util.as_numeric_if_number) %>%
+    util.apply_columns(., as_logical_if_logical)
+}
+
```

Comment:
Hardcoding missing values and separation symbols reduces the flexibility the function. I'm not sure what a good solution is, since you cannot just add a test of this assumption is correct. May be have these as default parameters?

Reply:
yeah, good point. We could also just not include the `na.strings = c("NA", "")` parameter at all, and maybe change the function name to `util.fread_csv` so that the `sep = ","` makes more sense...the `colClasses` one is intrinsic to the function's purpose, so that one would need to stay.

Reply:
+1 to Sarah's idea